app/scripts/controllers/transactions/tx-state-manager.js

```diff
@@ -152,12 +152,15 @@ class TransactionStateManager extends EventEmitter {
     // or rejected tx's.
     // not tx's that are pending or unapproved
     if (txCount > txHistoryLimit - 1) {
-      const index = transactions.findIndex((metaTx) => {
-        return getFinalStates().includes(metaTx.status)
-      })
-      if (index !== -1) {
-        transactions.splice(index, 1)
-      }
+      do {
+        const index = transactions.findIndex((metaTx) => {
+          return getFinalStates().includes(metaTx.status)
+        })
+        if (index !== -1) {
+          transactions.splice(index, 1)
+        }
```

Comment:
Would it be correct to break in the `else` case here? As a safeguard against this looping infinitely - which would be impossible right now _I think_, but it's conceivable how this could happen with some future change.

Reply:
look a couple lines down maybe